EasyVCR.Tests/CensorsTest.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            const string body = "text here";
+            const InternalUtilities.ContentType contentType = InternalUtilities.ContentType.Text;
+
+            var censors = new Censors();
+
+            var result = censors.ApplyBodyParametersCensors(body, contentType);
+
+            Assert.AreEqual(body, result);
+        }
+
+        [TestMethod]
+        public void TestApplyBodyParametersCensorsHtmlContentTypeReturnsOriginalString()
+        {
+            const string body = "html code here";
+            const InternalUtilities.ContentType contentType = InternalUtilities.ContentType.Html;
+
+            var censors = new Censors();
+
+            var result = censors.ApplyBodyParametersCensors(body, contentType);
+
+            Assert.AreEqual(body, result);
+        }
+
+        [TestMethod]
+        public void TestApplyBodyParametersCensorsXmlContentTypeReturnsOriginalString()
+        {
+            const string body = "xml code here";
+            const InternalUtilities.ContentType contentType = InternalUtilities.ContentType.Xml;
+
+            var censors = new Censors();
+
+            var result = censors.ApplyBodyParametersCensors(body, contentType);
+
+            Assert.AreEqual(body, result);
+        }
+
+        [TestMethod]
+        public void TestApplyHeaderCensorsNoHeadersReturnsOriginalHeaders()
+        {
+            var headers = new Dictionary<string, string> { };
+
+            var censors = new Censors();
+
+            var result = censors.ApplyHeaderCensors(headers);
+
+            Assert.AreEqual(headers, result);
+        }
+
+        [TestMethod]
+        public void TestApplyHeaderCensorsNoCensorsReturnsOriginalHeaders()
+        {
+            var headers = new Dictionary<string, string> { { "foo", "bar" } };
+
+            var censors = new Censors();
+
+            var result = censors.ApplyHeaderCensors(headers);
+
+            Assert.AreEqual(headers, result);
+        }
+
+        [TestMethod]
+        public void TestApplyUrlCensorsNoUrlReturnsOriginalUrl()
+        {
+            string? url = null;
+
+            var censors = new Censors();
+
+            var result = censors.ApplyUrlCensors(url);
+
+            Assert.AreEqual(url, result);
+        }
+
+        [TestMethod]
+        public void TestApplyUrlCensorsNoQueryOrPathCensorsReturnsOriginalUrl()
+        {
+            const string url = "some url";
+
+            var censors = new Censors();
+
+            var result = censors.ApplyUrlCensors(url);
+
+            Assert.AreEqual(url, result);
+        }
+
+        [TestMethod]
+        public void TestApplyPathElementsCensorsNoUrlReturnsOriginalUrl()
+        {
+            string? url = null;
+
+            var censors = new Censors();
+
+            var result = censors.ApplyPathElementsCensors(url);
+
+            Assert.AreEqual(url, result);
+        }
+
+        [TestMethod]
+        public void TestApplyPathElementsCensorsNoCensorsReturnsOriginalUrl()
```

Comment:
question: we appear to be testing a bunch of the "empty" to "original" flows, are we already testing the "filled" flows elsewhere?

Reply:
Yes, we use censors in the unit tests for other functionality, which doesn't seem to interfere with their operation.

This is a bunch of happy and sad path testing for censors specifically.